reporters/reporter-default/src/main/java/org/radargun/reporting/html/TimelineDocument.java

```diff
@@ -188,81 +182,95 @@ public Void call() throws Exception {
             log.error("Failed to generate on of the charts: ", e);
          }
       }
+   }
 
-      /* Checkboxes */
-      write("<div style=\"float: left;\">\n");
-      for (Map.Entry<String, Integer> eventEntry : eventCategories.entrySet()) {
-         write(String.format("<input id=\"cat_%d\" type=\"checkbox\" checked=\"checked\" onClick=\"", eventEntry.getValue()));
+   /**
+    * The following methods are used in Freemarker templates
+    * e.g. method getPercentiles() can be used as getPercentiles() or percentiles in template
+    */
 
-         for (Timeline timeline : timelines) {
-            for (int valuesId : valueCategories.values())  {
-               write(String.format("reset_display('layer_%d_%d_%d', this.checked && is_checked('slave_%d'), 'block');",
-                     valuesId, eventEntry.getValue(), timeline.slaveIndex, timeline.slaveIndex));
-            }
-         }
-         write(String.format("\"><strong>%s</strong><br>\n", eventEntry.getKey()));
-      }
+   public Configuration getConfiguration() {
+      return configuration;
+   }
+
+   public String getConfigName() {
+      return configName;
+   }
 
-      write("<br><br>");
-      List<Cluster.Group> groups = cluster.getGroups();
+   public Cluster getCluster() {
+      return cluster;
+   }
+
+   public List<Timeline> getTimelines() {
+      return timelines;
+   }
+
+   public Map<String, Double> getMinValues() {
+      return minValues;
+   }
+
+   public Map<String, Double> getMaxValues() {
+      return maxValues;
+   }
+
+   public Map<String, Integer> getValueCategories() {
+      return valueCategories;
+   }
+
+   public Map<String, Integer> getEventCategories() {
+      return eventCategories;
+   }
+
+   public long getStartTimestamp() {
+      return startTimestamp;
+   }
+
+   public long getEndTimestamp() {
+      return endTimestamp;
+   }
+
+   public String generateEventChartFile(int eventCategoryId, int slaveIndex) {
+      return String.format("timeline_%s_e%d_%d.png", configName, eventCategoryId, slaveIndex);
+   }
+
+   public String resetDisplay(Integer value) {
```

Comment:
I feel this is still kind of template specific & could be moved to template. Not sure how much effort would that require, wdyt?